### PR TITLE
Re-implement support for "phrase" matching, handle display of single quotes in search terms

### PR DESCRIPTION
**Pull request by hancush**
## Overview

I goofed up handling of phrases enclosed in quotes when I added the SES integration. This PR adds back proper handling of exact phrases. In summary, it:

- Filters the default queryset for results containing all search terms (e.g., if you search for "metro red line" and "train", you'll only get results containing both terms)
- If searching attachment text, also return results where the attachment text contains both terms
- Repairs the display of single quotes when rendering a pre-existing search term
- Handles quoted search terms when highlighting topics matching the query

### Checklist

- [x] PR has a descriptive enough title to be useful in changelogs

### Notes

I'm going to deploy this to the staging site for further QA.

N.b., this PR does _not_ handle single quote parsing on the back end. That is a deeper issue that I'd like to address separately.

## Testing Instructions

Perform the following searches on the staging site:

* Search `metro red line` and confirm you see results, not necessarily containing the full phrase. Toggle to tag and keyword results and confirm you see the correct results.
* Search `"metro red line"` and confirm you only see results containing the full phrase.
* Search `red line` and confirm you see results. Toggle to the keyword results and confirm you see results tagged `Metro Red Line`.
* Search `it's off limits` and confirm the search term is rendered correctly (i.e., no escaped characters) in the search bar.

Handles #640, handles #663.

### CODE REVIEW

File: lametro/templates/search/search.html [hunk 1]
```diff
@@ -298,7 +298,7 @@ <h3 class="modal-links">
     var queryTerms = '{{ request.GET.q }}'
         .split(' AND ')
         .reduce(function (terms, query) {
-            var formattedQuery = query.replace(/&quot;/g, '"');
+            var formattedQuery = query.replace(/&quot;/g, '"').replace(/&#39;/g, "'");
```
Comment (hancush) [inline lametro/templates/search/search.html:301]: Unescape both double and single quotes.

File: lametro/templatetags/lametro_extras.py [hunk 2]
```diff
@@ -176,12 +176,28 @@ def get_highlighted_attachment_text(context, id):
 @register.filter
 def matches_query(tag, request):
     # request.GET['q'] looks like "token AND token AND token"
-    return tag.lower() in [token.lower() for token in request.GET.get('q', '').split(' AND ')]
+    terms = [
```
Comment (hancush) [inline lametro/templatetags/lametro_extras.py:179]: The changes to the template tags file highlight tags that both exactly and partially match search terms, e.g., searching "red line" will highlight "metro red line".

File: lametro/views.py [hunk 3]
```diff
@@ -610,43 +610,55 @@ def __init__(self, *args, **kwargs):
...
+            sqs = sqs.exclude(id__in=tagged_results)
 
         elif self.result_type == 'topic':
-            sqs = sqs.filter(tag_filter)
+            sqs = sqs.filter(id__in=tagged_results)
+
+        return sqs
+
+    def _identifier_search(self, sqs):
+        pass
```
Comment (hancush) [inline lametro/views.py:652]: Will implement this in #487.

File: lametro/views.py [hunk 4]
```diff
@@ -610,43 +610,55 @@ def __init__(self, *args, **kwargs):
...
+        report_filter = SQ()
+        attachment_filter = SQ()
 
-        if q:
-            return ' AND '.join('({})'.format(term.strip()) for term in q.split('AND'))
-        else:
-            return ''
+        for token in self.cleaned_data['q'].split(' AND '):
+            report_filter &= SQ(text=Raw(token))
+            attachment_filter &= SQ(attachment_text=Raw(token))
```
Comment (hancush) [inline lametro/views.py:619]: Most of this is reorganization. This is the main change: I revert back to issuing a `Raw` search for each term in the query, so double quotes are preserved. https://django-haystack.readthedocs.io/en/master/inputtypes.html#raw

**Review by fatima3558 — APPROVED** [review]
The search terms about the metro red line work as described, and the `it's off limits` term renders correctly in the search bar. Looking good!

**Pull request merged**